Libraries/LibGUI/VimEditingEngine.cpp

```diff
@@ -107,6 +107,12 @@ bool VimEditingEngine::on_key_in_normal_mode(const KeyEvent& event)
                 move_one_up(event);
                 switch_to_insert_mode();
                 break;
+            case (KeyCode::Key_LeftBrace):
+                move_to_previous_empty_lines_block();
+                break;
+            case (KeyCode::Key_RightBrace):
+                move_to_next_empty_lines_block();
+                break;
```

Comment:
This relies too much on the keyboard layout in my opinion.
Can't we use `event.code_point() == '{'` et cetera here?

Reply:
That seems worse. What does it mean to "rely too much on the keyboard layout"?

Reply:
If (read: when) at some point we decide to split this off `TextEditor`, and `LibGUI` in general (like my original suggestion), either these key events will have to go, or we'll have to make fake key events somehow.
`KeyCode` seems better than raw characters though, I say we keep that but get rid of the key event?

Reply:
I mean that this will only behaves like vim  if `{` is `shift + [` (and `ctrl` and `alt` are not held down).

Also, the other bindings that use an uppercase letter (e.g. `O`) will only work if you press `shift + o` and not if you only press `o` with CapsLock activated (unlike vim).